apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -0,0 +1,98 @@
+defmodule GameBackend.CurseOfMirra.Quests do
+  @moduledoc """
+    Module to work with quest logic
+  """
+  alias GameBackend.Quests.DailyQuest
+  alias GameBackend.Repo
+  alias GameBackend.Quests.Quest
+  alias Ecto.Multi
+  import Ecto.Query
+
+  @doc """
+  Get a %Quest{} by the config_id field
+
+  ## Examples
+
+      iex>get_quest_by_config_id(4)
+      %Quest{config_id: 4}
+
+  """
+  def get_quest_by_config_id(quest_config_id) do
+    Repo.get_by(Quest, config_id: quest_config_id)
+  end
+
+  @doc """
+  Get all %Quest{} by the type field
+
+  ## Examples
+
+      iex>get_quest_by_config_id("daily")
+      [%Quest{type: "daily"}]
+  """
+  def get_quests_by_type(type) do
+    q =
+      from(qd in Quest,
+        where: qd.type == ^type
+      )
+
+    Repo.all(q)
+  end
+
+  @doc """
+  Run the quest changeset with the given attrs
+  Return %Changeset{}
+
+  ## Examples
+
+      iex>change_quest(quest, attrs)
+      %Changeset{}
+
+  """
+  def change_quest(quest, attrs) do
+    Quest.changeset(quest, attrs)
+  end
+
+  @doc """
+  Insert or update config quests present in the "quests_descriptions.json" file
+  """
+  def upsert_quests(quests_params) do
```

Comment:
If you feel curious you can try the actual [Ecto Upsert](https://hexdocs.pm/ecto/constraints-and-upserts.html#upserts) rather than implement your own, but full disclosure I have never used it so no idea if it will work, but would be a nice improvement here if it does

Reply:
hmmm if we use the Ecto upsert we should update every other upsert already implemented in the repo, maybe we could do this change in a proper pr